### ISSUE TITLE
Testability strategy

### ISSUE THREAD

**Issue by ayr-ton** [gogits/gogs#2293]
Hey there.

I was digging into the code and saw a excellent piece of work from all the contributors everywhere.

However, something caught my attention. The code base is growing every day but there's no Unit Tests mention at the CONTRIBUTING.md file. As Go has a good built-in support for automated tests (https://golang.org/pkg/testing/) I would like to ask if you guys have plans to design a testing strategy as something required for future contributions. 

It could increase confidence in recommending the project for production use as well, as I think gogs could replace a lot of solutions out there.

Also, I can see there's a few modules with tests https://github.com/gogits/gogs/search?p=1&q=testing&utf8=%E2%9C%93 but the coverage still small.

If you :+1: this idea, we can discuss here the strategy.

**Comment by Unknwon** [comment]
Yes, unit tests can only be applied to small part of Gogs's codebase.

I'm waiting to finish APIs to achieve more on testing.

**Comment by Unknwon** [comment]
Also, I suggest move discussion to http://forum.gogs.io/

**Comment by Unknwon** [comment]
ping @ayr-ton ?

**Comment by ayr-ton** [comment]
Pong @Unknwon 
I will be moving the discussion to the forum and then I will be posting the results here :)

**Comment by ayr-ton** [comment]
Related post:
http://forum.gogs.io/topic/21/testability-strategy

**Comment by tebanep** [comment]
Happened exactly the same to me, when looking for tests. Hope an strategy can be set so the confidence in the program increases. Though, I'm using gogs in production right now without problems. Thanks for it!
